vector/tests/Utilities.hs

```diff
@@ -121,10 +121,24 @@ instance TestData ty where { \
 id_TestData(())
 id_TestData(Bool)
 id_TestData(Int)
-id_TestData(Float)
-id_TestData(Double)
 id_TestData(Ordering)
 
+instance TestData Float where
+  type Model Float = Float
+  model = id
+  unmodel = id
+
+  type EqTest Float = Property
+  equal x y = property (x == y || isNaN x == isNaN y)
```

Comment:
```suggestion
  equal x y = property (x == y || (isNaN x && isNaN y))
```
With your definition, every non-NaN number is equal to every other non-NaN number.

Reply:
Thanks a lot!